src/azul/indexer/log_forwarding_service.py

```diff
@@ -19,59 +26,135 @@
 )
 
 
-class LogForwardingService:
+class LogForwardingService(abc.ABC):
+
+    def read_logs(self, bucket: str, key: str) -> Iterator[MutableJSON]:
+        response = aws.s3.get_object(Bucket=bucket, Key=key)
+        body = self._read_lines(response['Body'])
+        return self._read_logs(body)
+
+    def _read_logs(self, file_body: Iterable[str]) -> Iterator[MutableJSON]:
+        # CSV format escapes the quotechar by repeating it. This cannot
+        # occur in the logs because quotations marks occurring within the
+        # field values are escaped. AWS does not document how the access
+        # logs are encoded, but our experiments indicate that characters
+        # including quotation marks, backslashes, and non-ASCII characters
+        # are escaped using a syntax based on NGINX log format
+        # (http://nginx.org/en/docs/http/ngx_http_log_module.html#log_format).
+        # For example, quotation marks are represented as `\x22`.
+        # FIXME vwrify that character-escaping works the same way for S3
```

Comment:
```suggestion
        # FIXME verify that character-escaping works the same way for S3
```

Typo. Also, I can't find documentation for it, but it's been standard practice to include a link to the related GitHub issue in the FIXME comment.